src/rebar_prv_install_deps.erl

```diff
@@ -239,81 +249,77 @@ package_to_app(DepsDir, Packages, {Name, Vsn}) ->
-                                                           ,StateAcc1
-                                                           ,LocksAcc)
-                                         end,
-                                     {SrcDepsAcc1, PkgDepsAcc1, SrcAppsAcc1, StateAcc2, SeenAcc1, LocksAcc1}
-                             end
-                     end,
-                     {[], PkgDeps, SrcApps, State, Seen, Locks},
-                     rebar_utils:sort_deps(SrcDeps)) of
+    case lists:foldl(
+            fun(AppInfo, {SrcDepsAcc, PkgDepsAcc, SrcAppsAcc, StateAcc, SeenAcc, LocksAcc}) ->
+                    update_src_dep(AppInfo, Profile, Level,
+                                   SrcDepsAcc, PkgDepsAcc, SrcAppsAcc, StateAcc,
+                                   Upgrade, SeenAcc, Locks, LocksAcc)
+            end,
+            {[], PkgDeps, SrcApps, State, Seen, Locks},
+            rebar_utils:sort_deps(SrcDeps)) of
         {[], NewPkgDeps, NewSrcApps, State1, Seen1, _NewLocks} ->
             {State1, NewSrcApps, NewPkgDeps, Seen1};
         {NewSrcDeps, NewPkgDeps, NewSrcApps, State1, Seen1, NewLocks} ->
             update_src_deps(Profile, Level+1, NewSrcDeps, NewPkgDeps, NewSrcApps, State1, Upgrade, Seen1, NewLocks)
     end.
 
+update_src_dep(AppInfo, Profile, Level, SrcDeps, PkgDeps, SrcApps, State, Upgrade, Seen, BaseLocks, Locks) ->
+    %% If not seen, add to list of locks to write out
+    Name = rebar_app_info:name(AppInfo),
+    case sets:is_element(Name, Seen) of
+        true ->
+            update_seen_src_dep(AppInfo, Level,
+                                SrcDeps, PkgDeps, SrcApps,
+                                State, Upgrade, Seen, BaseLocks, Locks);
+        false ->
+            update_unseen_src_dep(AppInfo, Profile, Level,
+                                  SrcDeps, PkgDeps, SrcApps,
+                                  State, Upgrade, Seen, Locks)
+
+    end.
+
+update_seen_src_dep(AppInfo, Level, SrcDeps, PkgDeps, SrcApps, State, Upgrade, Seen, BaseLocks, Locks) ->
+    Name = rebar_app_info:name(AppInfo),
+    %% If seen from lock file don't print warning about skipping
+    case lists:keymember(Name, 1, BaseLocks) of
+        false ->
+            warn_skip_deps(AppInfo);
+        true ->
+            ok
+    end,
+    %% scan for app children here if upgrading
+    case Upgrade of
+        false ->
+            {SrcDeps, PkgDeps, SrcApps, State, Seen, Locks};
+        true ->
+            {NewSrcDeps, NewPkgDeps, NewSrcApps, NewState, NewLocks}
+            = handle_dep(AppInfo, SrcDeps, PkgDeps, SrcApps,
+                         Level, State, Locks),
+            {NewSrcDeps, NewPkgDeps, NewSrcApps, NewState, Seen, NewLocks}
+    end.
+
+update_unseen_src_dep(AppInfo, Profile, Level, SrcDeps, PkgDeps, SrcApps, State, Upgrade, Seen, Locks) ->
+    {NewSeen, State1} = maybe_lock(Profile, AppInfo, Seen, State, Level),
+    {NewSrcDeps, NewPkgDeps, NewSrcApps, State2, NewLocks}
+    = case Upgrade of
```

Comment:
I have a nit pick on this, being on the same column as the start of the expression above should mean it is a new expression. If it isn't it should be indented :)